src/Datadog.Trace/Sampling/RegexSamplingRule.cs

```diff
@@ -0,0 +1,143 @@
+                    {
+                        var kvp = rulePart.Split(new[] { "=" }, StringSplitOptions.None);
+
+                        if (kvp.Length != 2 || string.IsNullOrWhiteSpace(kvp[0]) || string.IsNullOrWhiteSpace(kvp[1]))
+                        {
+                            Log.Warning("Rule {0} is malformed, skipping.",  ruleName);
+                            continue;
+                        }
+
+                        var key = kvp[0].Trim();
+                        var value = kvp[1].Trim();
+
+                        if (key.Equals("rate", StringComparison.OrdinalIgnoreCase) && float.TryParse(value, out rate))
+                        {
+                            if (rate < 0 || rate > 1)
+                            {
+                                // invalid rate
+                                Log.Warning("Invalid rate {0} specified for sampling rule {1}, skipping.", rate, ruleName);
+                                break;
+                            }
+
+                            rateSet = true;
+                        }
+                        else if (key.Equals("service", StringComparison.OrdinalIgnoreCase))
+                        {
+                            serviceNameRegex = WrapWithLineCharacters(value);
+                        }
+                        else if (key.Equals("operation", StringComparison.OrdinalIgnoreCase))
+                        {
+                            operationNameRegex = WrapWithLineCharacters(value);
+                        }
+                        else if (key.Equals("name", StringComparison.OrdinalIgnoreCase))
+                        {
+                            ruleName = value;
+                        }
+                    }
+
+                    if (rateSet == false)
+                    {
+                        // Need a valid rate to be set to use a rule
+                        Log.Warning("Rule {0} is missing the required rate, skipping.",  ruleName);
+                        continue;
+                    }
+
+                    yield return new RegexSamplingRule(
+                        rate: rate,
+                        name: ruleName,
+                        serviceNameRegex: serviceNameRegex,
+                        operationNameRegex: operationNameRegex);
+                }
+            }
+        }
+
+        public bool IsMatch(Span span)
+        {
+            if (_serviceNameRegex != null && !Regex.IsMatch(input: span.ServiceName, pattern: _serviceNameRegex))
+            {
+                return false;
+            }
+
+            if (_operationNameRegex != null && !Regex.IsMatch(input: span.OperationName, pattern: _operationNameRegex))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public float GetSamplingRate()
+        {
+            return _samplingRate;
+        }
+
+        private static string WrapWithLineCharacters(string regex)
```

Comment:
This seems unexpected, what if I wanted to do a regex for something in the middle?

Reply:
`.*cart.*`

Reply:
[See previous discussion here](https://github.com/DataDog/dd-trace-dotnet/pull/537/files#r339699459).

Reply:
@lucaspimentel your link doesn't link to anything.

Reply:
Uhm, maybe because the comment is marked as resolved? 🤦‍♂ It's a comment I made on `ConfigurationKeys.cs`, line 107. Maybe [this link to that line](https://github.com/DataDog/dd-trace-dotnet/pull/537/files#diff-8690a9e9e2b8330913d452daaa4b27cfR107) works better.

Just copying it here:

> Regex note: the example service=cart.* matches services that contain cart anywhere in the name (e.g. `my-cartoon`). To make it match service names that start with cart, users can use something like `^cart`. (The `.*` at the end is redundant.)
> 
> Similarly, `operation=web` will match any service with `web` anywhere in the name (e.g. `mywebapp`). To get an exact match, users will need to use `operation=^web$`.
> 
> Maybe we should add the `^` and `$` ourselves to every regex? That way a setting like `prefix-.*` or `.*-suffix` [or `web`] will work as the user intended?

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/537#discussion_r339699459